gemac/management.py

```diff
@@ -0,0 +1,155 @@
+from myhdl import block, Signal, intbv, always_seq, always_comb
+from myhdl._ShadowSignal import TristateSignal
+
+
+rx0, rx1, tx, flow, managementreg, ucast0, \
+    ucast1, addrtable0, addrtable1, addrfiltermode = range(10)
+hostclk_count = 0
+
+
+class mdioData:
+    def __init__(self, opcode, hostaddr, wrdata):
+        self.pre = intbv(0xFFFFFFFF)[32:]
+        self.st = intbv(0b01)[2:]
+        self.op = opcode[2:]
+        self.phyadd = hostaddr[10:5]
+        self.regadd = hostaddr[5:]
+        self.wrdata = wrdata[16:]
+
+
+@block
+def management(host_interface, mdio_interface):
+
+    """
+    Management Data - Input Output Interface.
```

Comment:
You should have more comments describing what this ~~module~~ block does.  I assume this module converts host transactions to MII MDIO transactions?  Also, you should consider "host"-less applications!

From my perspective this is important, a user should be able to give a complete configuration at design time and the gemac core should be operational without an embedded processor.